modules/core/src/main/java/org/apache/ignite/internal/processors/datastructures/GridCacheCountDownLatchImpl.java

```diff
@@ -236,6 +236,9 @@ public GridCacheCountDownLatchImpl(String name,
     @Override public void onUpdate(int cnt) {
         assert cnt >= 0;
 
+        if (initGuard.get())
```

Comment:
Add comments to this code to make it clear for other developers why this code is there.

It can sound this way - "if the latch is being initialized on this node as well we have to wait for initialization completion to avoid possible race conditions"